### ISSUE TITLE
error using the default base.py from TPGSR

### ISSUE THREAD

**Issue by leocd91** [zhaominyiz/C3-STISR#7]
hey, it seems if we want to run the test/train using your arch, we need to modify the `base.py` from `interface` directory (especially the 293rd line about `generator_init`.
can you share the modified base.py?

**Comment by zhaominyiz** [comment]
My base.py is in a mass. I have defined a lot of new models. Therefore, it will be a bit difficult for me to release them. Following are some key functions:
# Define the model and loss function
```python
elif self.args.arch == 'c3stisr':
      model = c3.TSRN_TL(scale_factor=self.scale_factor, width=cfg.width, height=cfg.height,
                           STN=self.args.STN, mask=self.mask, srb_nums=self.args.srb,
                           hidden_units=self.args.hd_u,triple_clues=True)
      print("Build Text Focused Loss")
      image_crit = text_focus_loss.TextFocusLoss(self.args)
```
# used for the last stage fine-tuning:
```python
params = torch.load('ckpt/C3-STISR-FIX/model_best_0.pth')['state_dict_G']
filtered_dict = {}
# reinit LM
for k in params.keys():
    pre_fix_name = k.split('.')[0]
    if pre_fix_name in ['correction_model']:
        continue
    filtered_dict['module.'+k] = params[k]
params = torch.load('ckpt/BCN_correct_model.pt')
for k in params.keys():
    filtered_dict['module.correction_model.'+k] = params[k]
model.load_state_dict(filtered_dict, strict=False)
```

**Comment by zhaominyiz** [comment]
If you still fail to run the code, I can offer you the full code secretly via email. But do not distribute them :)

**Comment by zhuzhiqiang148** [comment]
@zhaominyiz I met the same issue,could you provide me the full code?I want to learn more about this aspect.My email :1487458940@qq.com

**Comment by zhaominyiz** [comment]
> @zhaominyiz I met the same issue,could you provide me the full code?I want to learn more about this aspect.My email :[1487458940@qq.com](mailto:1487458940@qq.com)

I have attached the base.py through the email. Please check it and if oky, how about give me a star or '一键三连'

**Comment by zhuzhiqiang148** [comment]
thanks,i have given you a star!

**Comment by zhuzhiqiang148** [comment]
OK.I also want to ask how long the training will be completed with only one Nvidia RTX 3090?




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月29日(星期二) 晚上6:42
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [zhaominyiz/C3-STISR] error using the default base.py from TPGSR (Issue #7)





 
Closed #7 as completed.

**Comment by zhaominyiz** [comment]
> OK.I also want to ask how long the training will be completed with only one Nvidia RTX 3090?
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2022年11月29日(星期二) 晚上6:42 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [zhaominyiz/C3-STISR] error using the default base.py from TPGSR (Issue #7) Closed #7 as completed. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

About 2 days for pre-training and half a day for finetuning.

**Comment by CongHuang96** [comment]
I have given my start!
 I also met the same issue, could you please provide me with the full code?
My email is t-conghuang@microsoft.com. 
Thanks a lot!
